ecommerce/backend/API/views/vendor.py

```diff
@@ -14,3 +14,20 @@ def vendor_details(request,vendor_id):
     if vendor is None:
         return Response(status=status.HTTP_400_BAD_REQUEST)
     return Response({'title':vendor.title,'image_url':vendor.image_url,'first_name':vendor.first_name,'last_name':vendor.last_name,'last_name':vendor.last_name,'total_rating':vendor.total_rating,'rating_count':vendor.rating_count,'description':vendor.description})
+
+@api_view(['PUT'])
+@permission_classes([permissions.IsVendorUser]) 
+def vendor_profile(request):
+    vendor = User.objects.filter(username=request.user).first()
```

Comment:
I am suspicious about this line of code. request.user is a user object. Checking its equality with username would fail I think. 

vendor = Vendor.objects.filter(user_id=request.user.id).first()

would be appropriate